### PR TITLE
Added initial version of hacking guide

### PR DESCRIPTION
**Pull request by richcarl**
An overview of the codebase to help programmers.

### CODE REVIEW

File: docs/hacking.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
+Hacking the Aeternity Codebase
+
+- Building
+  - See [Build](build.md)
+    - Ubuntu dependencies: `sudo apt install autoconf build-essential cmake
+      erlang libsodium-dev libgmp-dev`
+    - Mac OS: `brew install erlang@24 openssl libsodium autoconf gmp cmake
```
Comment (hanssv) [inline docs/hacking.md:7]: Picking the _oldest_ OTP-version we support is a sure way to make it outdated as quickly as possible 😅

**Comment by richcarl** [comment]
Can I get a couple of approvals so I can merge this? Then we can all help adding and modifying.

**Review by dincho — APPROVED** [review]
The bits on the relx and rebar3 seems to be outdated, we have copied and modified(?) some of the relx binaries/scripts I think.

**Comment by richcarl** [comment]
> The bits on the relx and rebar3 seems to be outdated, we have copied and modified(?) some of the relx binaries/scripts I think.

Let's investigate that and update the doc later in that case.

**Comment by dincho** [comment]
> > The bits on the relx and rebar3 seems to be outdated, we have copied and modified(?) some of the relx binaries/scripts I think.
> 
> Let's investigate that and update the doc later in that case.

While I did approved this PR, it doesn't make sense in general to commit an outdated info into master

**Comment by hanssv** [comment]
Wouldn't this info fit better in the wiki? Though that is probably even more outdated.

**Review by hanssv — APPROVED** [review]

**Pull request merged**